src/client/mod.rs

```diff
@@ -153,6 +161,21 @@ impl Client {
         Ok(client)
     }
 
+    /// Return an `EncryptedClientBuilder` for constructing a `Client` with auto-encryption enabled.
+    #[cfg(feature = "csfle")]
+    pub fn encrypted_builder(
+        options: ClientOptions,
+        key_vault_namespace: crate::Namespace,
+    ) -> EncryptedClientBuilder<false> {
```

Comment:
Could be nice to introduce aliases for these to better communicate what the const generic means, e.g.
```rust
pub type EncryptedClientBuilderNoKmsProviders = EncrypedClientBuilder<false>;
pub type EncryptedClientBuilderReady = EncrypedClientBuilder<true>;
```
These names aren't great but I imagine there could be good ones we could choose.